doc/api/path.md

```diff
@@ -166,12 +166,12 @@ path.extname('.index')
 
 A [`TypeError`][] is thrown if `path` is not a string.
 
-## path.format(pathObject)
+## path.format(options)
```

Comment:
Thanks for tackling this issue in the docs. I find the algorithmic explanation only marginally better than reading the code directly, so I'm glad to see an effort to improve this portion of the path doc.

The names of function arguments in the docs match the names in the code. So if we're going to change `pathObject` to `options` here, we probably want to change it in `lib/path.js` as well.

That said, I don't think `options` is better than `pathObject`. This is not an object containing options; it's an object containing data representing a path. I think `pathObject` is better.